web/src/main/scala/slamdata/engine/api/server.scala

```
@@ -108,6 +109,37 @@ object Server {
     }
   }
 
+  // NB: returns (), or else an explanation of why the port is not available,
+  // or fails if some other error occurs.
+  def available(port: Int): Task[String \/ Unit] = Task.delay {
```

> I'm handling the left on that line, so I don't see how propagating it to main helps anything.

Well, you're handling it exactly one time. As soon as you attempt binding to the port, if it fails a second time, that error will be lost; ideally the original error would propagate, which is something `ErrorT` machinery makes easy to do.

In addition, the monad in `main` should really be `ErrorT[E, Task, ?]`, because most of those effects can end in failures which will currently be swallowed into the `Task` `Throwable` black hole.

In any case, not a big deal since it's part of a bigger chance, happy to merge pending @sellout feedback.